crest/Assets/Crest/Crest/Scripts/Shapes/FFT/ShapeFFT.cs

```diff
@@ -398,12 +411,132 @@ public bool Validate(OceanRenderer ocean, ValidatedHelper.ShowMessage showMessag
+        /// </summary>
+        void BakeHelpBox(ShapeFFT target)
+        {
+            var message = "";
+
+            FFTBaker.ComputeRequiredOctaves(target._spectrum, target._smallestWavelengthRequired, out var smallestOctaveRequired, out var largestOctaveRequired);
+            if (largestOctaveRequired == -1 || smallestOctaveRequired == -1 || smallestOctaveRequired > largestOctaveRequired)
+            {
+                EditorGUILayout.HelpBox("No waves in spectrum. Increase one or more of the spectrum sliders.", MessageType.Error);
+                return;
+            }
+
+            message += $"FFT resolution is {target._resolution}.";
+            message += $" Spectrum power sliders give {largestOctaveRequired - smallestOctaveRequired + 1} active octaves greater than smallest wavelength {target._smallestWavelengthRequired}m.";
+            var scales = largestOctaveRequired - smallestOctaveRequired + 2;
+            message += $" Bake data resolution will be {target._resolution} x {target._resolution} x {scales}.";
+
+            message += "\n\n";
+            message += $"Period is {target._timeLoopLength}s.";
+            message += $" Frames per second setting is {target._timeResolution}.";
+            var frameCount = target._timeLoopLength * target._timeResolution;
+            message += $" Frame count is {target._timeLoopLength} x {target._timeResolution} = {frameCount}.";
+
+            message += "\n\n";
+            var pointsPerFrame = target._resolution * target._resolution * scales;
+            var channelCount = 4;
+            var bytesPerChannel = 4;
+            message += $"Total data size will be {pointsPerFrame * frameCount * channelCount * bytesPerChannel / 1048576f} MB.";
+
+            EditorGUILayout.HelpBox(message, MessageType.Info);
+        }
+
+        public override void OnInspectorGUI()
+        {
+            base.OnInspectorGUI();
+
+            var fft = target as ShapeFFT;
+
+            bool bakingEnabled = fft._enableBakedCollision;
+
+            if (bakingEnabled)
+            {
+                if (fft._spectrum == null)
+                {
+                    EditorGUILayout.HelpBox("A spectrum must be assigned to enable collision baking.", MessageType.Error);
+                    return;
+                }
+
+                BakeHelpBox(fft);
+            }
+
+            GUI.enabled = bakingEnabled;
+            OnInspectorGUIBaking();
+            GUI.enabled = true;
+        }
+
+        /// <summary>
+        /// Controls & GUI for baking.
+        /// </summary>
+        void OnInspectorGUIBaking()
+        {
+            if (OceanRenderer.Instance == null) return;
+
+            var bakeAndAssignLabel = "Bake to asset and assign to current settings";
+            var selectCurrentSettingsLabel = "Select current settings";
+            if (OceanRenderer.Instance._simSettingsAnimatedWaves != null)
+            {
+                if (GUILayout.Button(bakeAndAssignLabel))
```

Comment:
Should we disable this if Burst is not present? From testing it has caused crashing if data is used and wasn't baked using Burst.

Reply:
I could repro the crash when i first started testing, but after reinstalling latest burst (1.6.1) the crash may have gone away for me. I've tried various combinations of entering/exiting play mode, baking, with burst in/out and it doesnt seem to crash again.

Reply:
Excellent.